File: app/src/main/java/com/shakil/barivara/model/note/Note.java
package com.shakil.barivara.model.note;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.firebase.database.Exclude;

import java.util.HashMap;
import java.util.Map;

public class Note implements Parcelable {
    private String NoteId;
    private String Title;
    private String Description;
    private String Date;
    private String FireBaseKey;

    public Note() {
    }


    protected Note(Parcel in) {
        NoteId = in.readString();
        Title = in.readString();
        Description = in.readString();
        Date = in.readString();
        FireBaseKey = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(NoteId);
        dest.writeString(Title);
        dest.writeString(Description);
        dest.writeString(Date);
        dest.writeString(FireBaseKey);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Note> CREATOR = new Creator<Note>() {
        @Override
        public Note createFromParcel(Parcel in) {
            return new Note(in);
        }

        @Override
        public Note[] newArray(int size) {
            return new Note[size];
        }
    };

    @Exclude
    public Map<String, Object> toMap() {
        HashMap<String, Object> result = new HashMap<>();
        result.put("noteId", NoteId);
        result.put("title", Title);
        result.put("description", Description);
        result.put("date", Date);

        return result;
    }

    public String getNoteId() {
        return NoteId;
    }

    public void setNoteId(String noteId) {
        NoteId = noteId;
    }

    public String getTitle() {
        return Title;
    }

    public void setTitle(String title) {
        Title = title;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String description) {
        Description = description;
    }

    public String getDate() {
        return Date;
    }

    public void setDate(String date) {
        Date = date;
    }

    public String getFireBaseKey() {
        return FireBaseKey;
    }

    public void setFireBaseKey(String fireBaseKey) {
        FireBaseKey = fireBaseKey;
    }

    public static Creator<Note> getCREATOR() {
        return CREATOR;
    }
}

File: app/src/main/java/com/shakil/barivara/model/note/Note.kt
package com.shakil.barivara.model.note

import android.os.Parcel
import android.os.Parcelable
import com.google.firebase.database.Exclude

class Note : Parcelable {
    var noteId: String = ""
    var title: String = ""
    var description: String = ""
    var date: String = ""
    var fireBaseKey: String = ""

    constructor()
    constructor(note: Parcel) {
        noteId = note.readString() ?: ""
        title = note.readString() ?: ""
        description = note.readString() ?: ""
        date = note.readString() ?: ""
        fireBaseKey = note.readString() ?: ""
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(noteId)
        dest.writeString(title)
        dest.writeString(description)
        dest.writeString(date)
        dest.writeString(fireBaseKey)
    }

    @Exclude
    fun toMap(): Map<String, Any> {
        val result = HashMap<String, Any>()
        result["noteId"] = noteId
        result["title"] = title
        result["description"] = description
        result["date"] = date
        return result
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Note> {
        override fun createFromParcel(parcel: Parcel): Note {
            return Note(parcel)
        }

        override fun newArray(size: Int): Array<Note?> {
            return arrayOfNulls(size)
        }
    }
}
